src/crimson/osd/replicated_recovery_backend.h

```diff
@@ -47,16 +47,37 @@ class ReplicatedRecoveryBackend : public RecoveryBackend {
     Ref<MOSDPGRecoveryDelete> m);
   interruptible_future<> handle_recovery_delete_reply(
     Ref<MOSDPGRecoveryDeleteReply> m);
+  interruptible_future<PushOp> prep_push_to_replica(
+    const hobject_t& soid,
+    eversion_t need,
+    std::pair<pg_shard_t, pg_missing_t> pg_shard);
+  void calc_clone_subsets(
+    SnapSet& snapset, const hobject_t& soid,
+    const pg_missing_t& missing,
+    const hobject_t &last_backfill,
+    interval_set<uint64_t>& data_subset,
+    std::map<hobject_t, interval_set<uint64_t>>& clone_subsets);
+  void calc_head_subsets(
+    crimson::osd::ObjectContextRef obc,
+    SnapSet& snapset,
+    const hobject_t& head,
+    const pg_missing_t& missing,
+    const hobject_t &last_backfill,
+    interval_set<uint64_t>& data_subset,
+    std::map<hobject_t, interval_set<uint64_t>>& clone_subsets);
```

Comment:
Can we just return the data_subset and clone_subsets from calc_*_subsets?  Also, do these actually use any members of ReplicatedRecoveryBackend?  Can we make them free functions in another file and add unit tests for them specifically?